src/morale.cpp

```diff
@@ -348,7 +415,7 @@ void player_morale::decay( int ticks )
     std::for_each( points.begin(), points.end(), do_decay );
     remove_expired();
 
-    for( int i = 0; i < ticks; i++ ) {
+    for( int i = 0; i < ticks; ++i ) {
         update_bodytemp_penalty();
```

Comment:
Would be better if the function took `ticks` argument instead of being invoked in a loop.